### PR TITLE
Clears up some lints

### PR DESCRIPTION
**Pull request by Ben-PH**
This is branched off the resource flattening PR. Will merge once that's merged.

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ pub(crate) fn parse_params(input: &str) -> IResult<&str, Option<LocationParamete
                     LocationParamKeys::Submodules => params.set_submodules(Some(value.into())),
                     LocationParamKeys::Shallow => params.set_shallow(Some(value.into())),
                     LocationParamKeys::Arbitrary(param) => {
-                        params.add_arbitrary((param, value.into()))
+                        params.add_arbitrary((param, value.into()));
```
Comment (a-kenji) [inline src/parser.rs:49]: What lint is that? I can't reproduce that with:

```
cargo clippy
```

File: src/urls.rs [hunk 2]
```diff
@@ -61,11 +60,10 @@ impl UrlWrapper {
                 let segments = self
                     .url
                     .path_segments()
-                    .map(|c| c.collect::<Vec<_>>())
-                    .ok_or(NixUriError::Error(format!(
-                        "Error parsing from host: {}",
-                        input
-                    )))?;
+                    .map(std::iter::Iterator::collect::<Vec<_>>)
+                    .ok_or_else(|| {
```
Comment (a-kenji) [inline src/urls.rs:64]: :+1:

**Review by a-kenji — APPROVED** [review]
Generally :+1:, would be good to know how to trigger the lints here ;).

File: src/parser.rs (hunk 1 above)
Comment (Ben-PH) [reply src/parser.rs:49]: My bad, should have edited the PR comments. Basically this:

https://github.com/Ben-PH/nix-uri/blob/b702d95b3ff724298a63d2f21dc6fd59ad8f38eb/src/lib.rs#L1-L9

I've put one clippy warning per line, and added some `allow`s to take out the noise.

The idea being, is that by default, it's commented out. The normal dev process isn't affected by them, but every now and then, you can uncomment them, and go through things manually. 

If you run `cargo clippy -- -W clippy::pedantic, that will trigger this lint, I believe. alternatively, uncomment the warn-lines, and you should see the lint triggered.

**Review by a-kenji — APPROVED** [review]
:+1:

**Pull request merged**